cvat/apps/annotation/annotation.py

```diff
@@ -77,13 +77,58 @@ def serialize(self):
         if serializer.is_valid(raise_exception=True):
             return serializer.data
 
+    @staticmethod
+    def _is_shape_inside(shape, start, stop):
+        return start <= int(shape['frame']) <= stop
+
+    @staticmethod
+    def _is_track_inside(track, start, stop):
+        # should be a0 <= a1, b0 <= b1
+        def is_intervals_overlapped(a0, a1, b0, b1):
+            return 0 <= min(a1, b1) - max(a0, b0)
+
+        prev_shape = None
+        for shape in track['shapes']:
+            if prev_shape and not prev_shape['outside'] and \
+                is_intervals_overlapped(
+                    prev_shape['frame'], shape['frame'], start, stop):
+                    return True
+            prev_shape = shape
+
+        if not prev_shape["outside"] and prev_shape['frame'] <= stop:
+            return True
+
+        return False
+
+    @staticmethod
+    def _slice_track(track_, start, stop):
+        track = copy.deepcopy(track_)
+        segment_shapes = [s for s in track['shapes'] if AnnotationIR._is_shape_inside(s, start, stop)]
```

Comment:
@azhavoro , if we have a track with only one shape with the frame less than start, segment_shapes will be empty.

Reply:
this case will be handled later in 133 line

Reply:
Potentially in the line we can get first shape with outside == True.

Reply:
added filter to remove starting shapes with outside==True after processing